### DISCUSSION TITLE
Playoff Data

### DISCUSSION THREAD

**Discussion by pjockey** [MLB-LED-Scoreboard/mlb-led-scoreboard#558] [General]
Not sure what I'm missing, perhaps in the config.json, but now that my preferred team is out, all I'm getting is the date/time and weather, no playoff games or brackets.

What do I need to fix?

Thanks

**Comment by WardBrian** [comment]
The relevant settings are `rotation.only_preferred` and `standings.team_offday`/`standings.no_games_live`. Can you share some of your config file?

I saw the playoff bracket this morning, so I know it works this season

**Comment by pjockey** [comment]
```
        "standings": {
                "team_offday": true,
                "mlb_offday": true,
                "always_display": false,
                "display_no_games_live": true
        },
        "rotation": {
                "enabled": true,
                "scroll_until_finished": true,
                "only_preferred": false,
                "only_live": true,
                "rates": {
                        "live": 15.0,
                        "final": 15.0,
                        "pregame": 15.0
                },
                "while_preferred_team_live": {
                        "enabled": false,
                        "during_inning_breaks": false

```

**Comment by WardBrian** [comment]
Here's my config:

```
  "news_ticker": {
    "team_offday": false,
    "always_display": false,
    "preferred_teams": true,
    "traderumors": true,
    "mlb_news": true,
    "countdowns": true,
    "date": true,
    "date_format": "%A, %B %d",
    "display_no_games_live": false
  },
  "standings": {
    "team_offday": false,
    "mlb_offday": false,
    "always_display": false,
    "display_no_games_live": true
  },
  "rotation": {
    "enabled": true,
    "scroll_until_finished": true,
    "only_preferred": false,
    "only_live": true,
    "rates": {
      "live": 15.0,
      "final": 15.0,
      "pregame": 15.0
    },
    "while_preferred_team_live": {
      "enabled": true,
      "during_inning_breaks": true
    }
  },
```

**Reply by pjockey** [reply]
Changed mine to match yours. Game 2 of Tigers/Astros is showing.

Thanks

**Reply by c2mfj** [reply]
My scoreboard is only showing the news ticker and brackets, even during games, which does not mimic my regular season behavior. I believe my JSON is correct,.

{
  "preferred": {
    "teams": [
      "Braves"
    ],
    "divisions": [
      "NL East",
      "NL Wild Card",
      "AL Wild Card"
    ]
  },
  "news_ticker": {
    "team_offday": false,
    "always_display": false,
    "preferred_teams": true,
    "traderumors": true,
    "mlb_news": true,
    "countdowns": true,
    "date": true,
    "date_format": "%A, %B %-d",
    "display_no_games_live": false
  },
  "standings": {
    "team_offday": false,
    "mlb_offday": true,
    "always_display": false,
    "display_no_games_live": false
  },
  "rotation": {
    "enabled": true,
    "scroll_until_finished": true,
    "only_preferred": false,
    "only_live": false,
    "rates": {
      "live": 15.0,
      "final": 15.0,
      "pregame": 8.0
    },
    "while_preferred_team_live": {
      "enabled": false,
      "during_inning_breaks": false
    }
  },

**Comment by Charliethom** [comment]
I also have been seeing this same problem. I have 2 64x64 boards setup and at the start of the post season I was getting the live data and team play including the brackets. But since going into the second round of playoffs I get no live games, just the brackets. I have not changed anything in the config.json file. One setup is from last year, never touched, and I was getting live games through world series. Sounds more like problem is from where data is scorced from and not one problem on anyone's setup if it is configured right.

**Comment by WardBrian** [comment]
Can you share either or both configs?

**Comment by c2mfj** [comment]
Moving mine out of the reply to its own comment.

My scoreboard is only showing the news ticker and brackets, even during games, which does not mimic my regular season behavior. I believe my JSON is correct,.

{
"preferred": {
"teams": [
"Braves"
],
"divisions": [
"NL East",
"NL Wild Card",
"AL Wild Card"
]
},
"news_ticker": {
"team_offday": false,
"always_display": false,
"preferred_teams": true,
"traderumors": true,
"mlb_news": true,
"countdowns": true,
"date": true,
"date_format": "%A, %B %-d",
"display_no_games_live": false
},
"standings": {
"team_offday": false,
"mlb_offday": true,
"always_display": false,
"display_no_games_live": false
},
"rotation": {
"enabled": true,
"scroll_until_finished": true,
"only_preferred": false,
"only_live": false,
"rates": {
"live": 15.0,
"final": 15.0,
"pregame": 8.0
},
"while_preferred_team_live": {
"enabled": false,
"during_inning_breaks": false
}
},

**Comment by WardBrian** [comment]
It looks like there may be a problem with is_offday:

https://github.com/MLB-LED-Scoreboard/mlb-led-scoreboard/blob/master/data/schedule.py#L85C24-L85C42

I don't know why this setting is checking `["standings"]["display_no_games_live"]`. I think it is doing the wrong thing if that setting is false, but since I use it, I never noticed.

This branch _should_ fix it, but I need to make sure it doesn't have other, unintended, consequences: https://github.com/MLB-LED-Scoreboard/mlb-led-scoreboard/compare/master...fix/offday

**Reply by WardBrian** [reply]
@c2mfj did you get a chance to try that branch?

**Reply by c2mfj** [reply]
Just tried and so far looks good. I'll update tonight whether or not I experience any issues!

**Reply by c2mfj** [reply]
Looking good with the live games tonight as well.

**Reply by unwound29** [reply]
No current games, unless Dodges or Padres added as the first preferred team. Even in the new branch. Giving the screen flow a read-through but first time in this code so it'll take a bit of time!

Logs with Dodges & Padres added as 2nd & 3rd preferred.

DEBUG (21:40:18): Updating schedule for 2024-10-08 21:40:18.703465
DEBUG (21:40:19): Fetching data for game 775321
DEBUG (21:40:19): Game Data Refreshed: 2024/10/08/lanmlb-sdnmlb-1
DEBUG (21:40:19): Current game is 0 seconds behind
DEBUG (21:40:19): Pre: <Pregame 0x7faf722cd0> LAD @ SD; 9:08PM; Walker Buehler (0-0 27.00 ERA) vs Michael King (1-0 1.00 ERA); Forecast: Clear and 68Â° wind 5 mph, R To L; TV: ['ESPN Radio', 'FS1']
DEBUG (21:40:19): Live: <Scoreboard 0x7fb0084390> LAD (1) @ SD (3); Status: In Progress; Inning: (Number: 2; State: Bottom); B:0 S:0 O:0; P:Buehler; AB:Higashioka; Bases: [X][ ][X];
DEBUG (21:40:19): Final: <Postgame 0x7fb0084390> W: Unknown 0-0; L: Unknown 0-0; S: None (None)


DEBUG (21:40:22): Main has selected the offday information to refresh

**Reply by WardBrian** [reply]
@unwound29 share your config? Do you have `team_offday` set in either the news or standings sections?

**Reply by unwound29** [reply]
Hey Brian, thanks for being willing to take a look!

So far I can see I'm going through the elif on line 82 of main.py, and the elif on line 43 in the renderers/main.py, which sticks the screen into a rotation between news and standings.

I could swear all summer long though, I got a game rotation even on my team's off days, but I don't see a path to get to __render_gameday unless it's both **NOT** a league off day, and **NOT** your team's off day.

In fact, if I force that if block into __render_gameday, the game does appear on the board. **_Edit_** Without refreshing the game info properly.

 I think this offday logic might not be working as intended.
`elif screen == ScreenType.LEAGUE_OFFDAY or screen == ScreenType.PREFERRED_TEAM_OFFDAY:`
If LEAGUE_OFFDAY is false and PREFERRED_TEAM_OFFDAY is true, I think we need to look at **rotation.only_preferred** from the config file, to determine if it should rendering a game, or only news/standings?


"preferred": {
		"teams": ["Blue Jays"],
		"divisions": ["AL East", "AL Wild Card"]
	},
	"news_ticker": {
		"team_offday": true,
		"always_display": false,
		"preferred_teams": true,
		"display_no_games_live": true,
		"traderumors": true,
		"mlb_news": true,
		"countdowns": true,
		"date": true,
		"date_format": "%A, %B %-d"
	},
	"standings": {
		"team_offday": false,
		"mlb_offday": true,
		"always_display": false,
		"display_no_games_live": true
	},
	"rotation": {
		"enabled": true,
		"scroll_until_finished": true,
		"only_preferred": false,
		"only_live": true,
		"rates": {
			"live": 15.0,
			"final": 15.0,
			"pregame": 15.0
		},
		"while_preferred_team_live": {
			"enabled": false,
			"during_inning_breaks": false
		}

**Comment by WardBrian** [comment]
Because you have ` "team_offday": true,` in `news`, I think your behavior is the expected one, even on my branch. Note that it is `team` singular -- only the number one preferred team is considered by that setting

**Reply by unwound29** [reply]
I see what you're saying, that setting overrides the rotation settings. Will check on it during tonight's games. Very much appreciated!
